nextstrain_profiles/nextstrain/builds.yaml

```diff
@@ -2,10 +2,36 @@
 custom_rules:
   - workflow/snakemake_rules/export_for_nextstrain.smk
   - nextstrain_profiles/nextstrain/subsampling_ranges.smk
-  # - nextstrain_profiles/nextstrain/download_preprocessed.smk
 
-S3_BUCKET: "nextstrain-ncov-private"
+# These parameters are only used by the `export_for_nextstrain` rule and shouldn't need to be modified.
+# To modify the s3 _source_ bucket, specify this directly in the `inputs` section of the config.
+# P.S. These are intentionally set as top-level keys as this allows command-line overrides. 
+S3_DST_BUCKET: "nextstrain-ncov-private"
+S3_DST_COMPRESSION: "xz"
+S3_DST_ORIGINS: ["gisaid"]
 
+# NOTE for shepherds -- there are commented out inputs here, you can
+# uncomment them to start the pipeline at that stage.
+# E.g. if you uncomment `filtered` then the pipeline
+# will start by downloading that file and proceeding straight to
+# subsampling
+inputs:
+  gisaid:
+    metadata: "s3://nextstrain-ncov-private/metadata.tsv.gz"
+    sequences: "s3://nextstrain-ncov-private/sequences.fasta.gz"
```

Comment:
@jameshadfield @tsibley This change is what's causing the CI tests to fail. The CI test was setup to run a Nextstrain global build on the "example data" from the repo without downloading any external data. It does this by overriding the `sequences` and `metadata` values of the config on the command line:

```
nextstrain build . all_regions -j 2 --profile nextstrain_profiles/nextstrain --config sequences=data/example_sequences.fasta metadata=data/example_metadata.tsv active_builds=global
```

But now that there are nested inputs here with different names, we'll need to use a different approach to testing. One fix would be to allow overriding `inputs` to an empty string from the command line like so:

```
nextstrain build . all_regions -j 2 --profile nextstrain_profiles/nextstrain --config sequences=data/example_sequences.fasta metadata=data/example_metadata.tsv active_builds=global inputs=
```

To do this, we would need to change a couple of Snakemake lines from:

```
if "inputs" not in config:
```

To:

```
if not config.get("inputs"):
```

These minimal changes make the CI command work for me again locally. The question remains whether we want to continue supporting the `sequences` and `metadata` keys in the future even for this minimal test case.

Reply:
Here is the patch for all the changes I needed to make:

```diff
diff --git a/workflow/snakemake_rules/common.smk b/workflow/snakemake_rules/common.smk
index 4618e04..fe52456 100644
--- a/workflow/snakemake_rules/common.smk
+++ b/workflow/snakemake_rules/common.smk
@@ -49,7 +49,7 @@ def _get_path_for_input(stage, origin_wildcard):
     if not origin_wildcard:
         # No origin wildcards => deprecated single inputs (e.g. `config["sequences"]`) which cannot
         # be downloaded from remote resources
-        if "inputs" in config:
+        if config.get("inputs"):
             raise Exception("ERROR: empty origin wildcard but config defines 'inputs`")
         path_or_url = config[stage] if stage in ["metadata", "sequences"] else ""
         remote = False
@@ -97,14 +97,14 @@ def _get_unified_metadata(wildcards):
     then that file is returned. Else "results/combined_metadata.tsv" is returned
     which will run the `combine_input_metadata` rule to make it.
     """
-    if "inputs" not in config:
+    if not config.get("inputs"):
         return config["metadata"]
     if len(list(config["inputs"].keys()))==1:
         return _get_path_for_input("metadata", "_"+list(config["inputs"].keys())[0])
     return "results/combined_metadata.tsv"

 def _get_unified_alignment(wildcards):
-    if "inputs" not in config:
+    if not config.get("inputs"):
         return "results/filtered.fasta"
     if len(list(config["inputs"].keys()))==1:
         return _get_path_for_input("filtered", "_"+list(config["inputs"].keys())[0])
```

Reply:
Resolved by 73c3455 - thanks John!